lib/Sema/TypeChecker.cpp

```diff
@@ -415,6 +415,35 @@ void swift::performWholeModuleTypeChecking(SourceFile &SF) {
 #endif
 }
 
+bool swift::isDifferentiableProgrammingEnabled(SourceFile &SF) {
+  auto &ctx = SF.getASTContext();
+  // Return true if differentiable programming is explicitly enabled.
+  if (ctx.LangOpts.EnableExperimentalDifferentiableProgramming)
+    return true;
+  // Otherwise, return true iff the `_Differentiation` module is imported in
+  // the given source file.
+  bool importsDifferentiationModule = false;
+  for (auto import : namelookup::getAllImports(&SF)) {
+    if (import.second->getName() == ctx.Id_Differentiation) {
+      importsDifferentiationModule = true;
+      break;
+    }
+  }
+  return importsDifferentiationModule;
```

Comment:
Is this `true` for the `_Differentiation` module itself? It seems like it must be, otherwise the `_Differentiation` module wouldn't compile successfully. But it's a bit surprising that `namelookup::getAllImports` returns the current module.